qualang_tools/plot/plot.py

```diff
@@ -1,17 +1,296 @@
-"""Tools to help handling plots from QUA programs.
+"""Tools to help to handle plots from QUA programs.
 
 Content:
     - interrupt_on_close: Allows to interrupt the execution and free the console when closing the live-plotting figure.
 """
 
 
+from typing import List
+import numpy as np
+from scipy import signal
+import plotly.graph_objects as go
+from matplotlib import pyplot as plt
+from qm.QmJob import QmJob
+
+
+def plot_demodulated_data_2d(
+    x,
+    y,
+    I,
+    Q,
+    x_label: str = None,
+    y_label: str = None,
+    title: str = None,
+    amp_and_phase: bool = True,
+    fig=None,
+    plot_options: dict = None,
+):
+    """
+    Plot 2D maps of either 'I' and 'Q' or the corresponding amplitude and phase in a single figure.
+
+    :param x: 1D numpy array representing the x-axis.
+    :param y: 1D numpy array representing the y-axis.
+    :param I: 2D numpy array representing the 'I' quadrature.
+    :param Q: 2D numpy array representing the 'Q' quadrature.
+    :param x_label: name for the x-axis label.
+    :param y_label: name for the y-axis label.
+    :param title: title of the plot.
+    :param amp_and_phase: if True, plots the amplitude [np.sqrt(I**2 + Q**2)] and phase [signal.detrend(np.unwrap(np.angle(I + 1j * Q)))] instead of I and Q.

```

Comment:
```suggestion
    :param amp_and_phase: if `True` (default value), plots the amplitude [np.sqrt(I**2 + Q**2)] and phase [signal.detrend(np.unwrap(np.angle(I + 1j * Q)))] instead of I and Q.
```

Reply:
thanks